enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -14,6 +14,23 @@
     [toucan.db :as db]
     [toucan.hydrate :refer [hydrate]]))
 
+(defn collection-set-for-user
+  "Given an optional user ID, find the transitive set of all Collection IDs which are either:
+  (a) global (ie. no one's personal collection);
+  (b) owned by this user (when user ID is non-nil); or
+  (c) descended from one of the above."
+  [user-or-nil]
+  (let [roots    (db/select ['Collection :id :location :personal_owner_id] :location "/")
```

Comment:
```suggestion
  (let [roots    (db/select [Collection :id :location :personal_owner_id] :location "/")
```

Quoting models in Toucan is really just meant as an escape hatch to fix circular reference problems between different `metabase.model.*` namespaces... if you don't need them it's generally better not to use them and refer to the model directly. It's better because tools that build dependency graphs between namespaces or that do multiple-namespace reloading or usage finders won't pick it up if it's the quoted symbol. Also it's a little tiny bit faster if Toucan doesn't have to manually look up the model the hard way every time this is called

Reply:
Don't forget you can also `:require` metabase.models` and get all the models you need in one go. 

It pulls in a ton of stuff so it's **probably** better to require the model namespaces individually if you have the patience... it might make the SerDes command start like half a second faster. For that reason I try not to use it too much in production code. But in tests since we load all the namespaces anyway it makes no difference. So use `metabase.models` in tests as much as you please